software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNode.java

```diff
@@ -72,7 +72,12 @@
 
     ConfigKey<String> RIAK_CONF_ADDITIONAL_CONTENT = ConfigKeys.newStringConfigKey(
             "riak.riakConf.additionalContent", "Template file (in freemarker format) for setting up additional settings in the riak.conf file", "");
-
+    
+    // numberOpenFiles' default value (65536) is based on the Basho's recommendation - http://docs.basho.com/riak/latest/ops/tuning/open-files-limit/ 
+    @SetFromFlag("numberOpenFiles")
+    AttributeSensorAndConfigKey<Integer, Integer> RIAK_NUM_OPEN_FILES = ConfigKeys.newIntegerSensorAndConfigKey(
```

Comment:
Using a plain `ConfigKey<Integer>` is fine here. You need an `AttributeSensor` only if reading the value back from the machine.